src/de/mod.rs

```diff
@@ -734,19 +828,42 @@ impl fmt::Display for Error {
     }
 }
 
-/// Deserializes an instance of type `T` from bytes of JSON text
+/// Deserializes an instance of type `T` from bytes of JSON text, using the provided buffer to unescape strings
 /// Returns the value and the number of bytes consumed in the process
-pub fn from_slice<'a, T>(v: &'a [u8]) -> Result<(T, usize)>
+pub fn from_slice_using_string_unescape_buffer<'a, T>(
+    v: &'a [u8],
+    string_unescape_buffer: &mut [u8],
+) -> Result<(T, usize)>
 where
     T: de::Deserialize<'a>,
 {
-    let mut de = Deserializer::new(v);
+    let mut de = Deserializer::new(v, string_unescape_buffer);
     let value = de::Deserialize::deserialize(&mut de)?;
     let length = de.end()?;
 
     Ok((value, length))
 }
 
+/// Deserializes an instance of type `T` from bytes of JSON text
+/// Returns the value and the number of bytes consumed in the process
+pub fn from_slice<'a, T>(v: &'a [u8]) -> Result<(T, usize)>
+where
+    T: de::Deserialize<'a>,
+{
+    from_slice_using_string_unescape_buffer(v, &mut [0; 128])
```

Comment:
```suggestion
    from_slice_using_string_unescape_buffer(v, &mut [])
```

We can't be allocating unused 128 byte buffers on peoples stacks for no reason here.